include/eeprom/core/accessor.hpp

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <array>
+#include <cassert>
+#include <vector>
+
+#include "addresses.hpp"
+#include "common/core/bit_utils.hpp"
+#include "task.hpp"
+#include "types.hpp"
+
+namespace eeprom {
+namespace accessor {
+
+class AccessorBuffer {
```

Comment:
i'm kind of concerned with lifetimes here. `AccessorBuffer` only holds pointers to the data you pass in; is it possible that an `AccessorBuffer` instance can outlive the data it points to? can we ensure that doesn't happen somehow?

Reply:
I looked and it does not seem that there is a good way to detect that a pointer points to valid memory this unless you attempt to catch SYSSEGV which is discouraged because its supposed to be an unrecoverable error.